os/net/lwip/src/api/sockets.c

```diff
@@ -399,27 +403,41 @@ struct lwip_sock *get_socket(int s)
 }
 
 #ifdef CONFIG_NET_NETMON
-struct lwip_sock *get_lwip_sock_info(void)
+struct socketlist *get_lwip_sock_info(void)
 {
-	return sockets;
+	struct socketlist *list = sched_getsockets();
+	return list;
 }
 #endif
 /**
  * Same as get_socket but doesn't set errno
  *
  * @param s externally used socket index
+ * @param pid process id used to get the corresponding socketlist
+ *
  * @return struct lwip_sock for the socket or NULL if not found
  */
-struct lwip_sock *tryget_socket(int s)
+struct lwip_sock *tryget_socket(int s, pid_t pid)
 {
+	struct lwip_sock *sock;
+	struct socketlist list;
+	FAR struct tcb_s *tcb;
 	s -= LWIP_SOCKET_OFFSET;
-	if ((s < 0) || (s >= NUM_SOCKETS)) {
+
+	if ((s < 0) || (s >= CONFIG_NSOCKET_DESCRIPTORS)) {
 		return NULL;
 	}
-	if (!sockets[s].conn) {
+
+	tcb = sched_gettcb(pid);
```

Comment:
This is same with sched_getsockets()

Reply:
how it is same with sched_getsockets()?  because i am fetching corresponding tcb from the pid.